src/util/pelz_json_parser.c

```diff
@@ -155,6 +176,40 @@ int encrypt_parser(cJSON * json, charbuf * key_id, charbuf * data)
     pelz_log(LOG_ERR, "Missing required JSON key: enc_data_len.");
     return (1);
   }
+  /* This will be re-instated when signatures and certificates are no longer optional in JSON requests
+  else if (!cJSON_HasObjectItem(json, "request_sig_val"))
+  {
+    pelz_log(LOG_ERR, "Missing required JSON key: request_sig_val.");
+    return (1);
+  }
+  else if (!cJSON_HasObjectItem(json, "request_sig_val_len"))
+  {
+    pelz_log(LOG_ERR, "Missing required JSON key: request_sig_val_len.");
+    return (1);
+  }
+  else if (!cJSON_HasObjectItem(json, "requestor_cert_val"))
+  {
+    pelz_log(LOG_ERR, "Missing required JSON key: requestor_cert_val.");
+    return (1);
+  }
+  else if (!cJSON_HasObjectItem(json, "requestor_cert_val_len"))
+  {
+    pelz_log(LOG_ERR, "Missing required JSON key: requestor_cert_val_len.");
+    return (1);
+  }
+  */
+  else if (!cJSON_HasObjectItem(json, "request_sig_val"))
+  {
+    cJSON_AddItemToObject(json, "request_sig_val", cJSON_CreateString("null\n"));
```

Comment:
I don't think a "null" signature value should be the string "null\n". I think a NULL pointer or a charbuf with a NULL chars pointer and a zero length would be better.